onnxscript/rewriter/generic_pattern_test.py

```diff
@@ -450,138 +422,121 @@ def validate_rotary_mapping(g, match_result) -> bool:
-                perms.append(list(n.attribute[0].ints))
+                perms.append(list(n.attributes["perm"].value))
             perm = perms[0]
             new_perm = [0 for p in perm]
             for i, p in enumerate(perms[1]):
                 new_perm[i] = perm[p]
             match_result.add_kwargs("perm", new_perm)
             return True
 
-        def transpose_transpose_apply_pattern(x, perm=None):
+        # FIXME(justinchuby): Support matched result binding
+        def transpose_transpose_apply_pattern(perm=None):
             if perm is None:
-                return oh.make_function(
+                return onnx.helper.make_function(
                     "any",
                     "id",
                     ["X"],
                     ["Y"],
                     [
-                        oh.make_node("Identity", ["X"], ["Y"]),
+                        onnx.helper.make_node("Identity", ["X"], ["Y"]),
                     ],
-                    [oh.make_opsetid("", 18)],
+                    [onnx.helper.make_opsetid("", 18)],
                 )
-            return oh.make_function(
+            return onnx.helper.make_function(
                 "any",
                 "id",
                 ["X"],
                 ["Y"],
                 [
-                    oh.make_node("Transpose", ["X"], ["Y"], perm=perm),
+                    onnx.helper.make_node("Transpose", ["X"], ["Y"], perm=perm),
                 ],
-                [oh.make_opsetid("", 18)],
+                [onnx.helper.make_opsetid("", 18)],
             )
 
-        model = oh.make_model(
-            oh.make_graph(
+        model = onnx.helper.make_model(
+            onnx.helper.make_graph(
                 [
-                    oh.make_node("Transpose", ["X"], ["xt"], perm=[1, 2, 0]),
-                    oh.make_node("Transpose", ["xt"], ["Y"], perm=[1, 2, 0]),
+                    onnx.helper.make_node("Transpose", ["X"], ["xt"], perm=[1, 2, 0]),
+                    onnx.helper.make_node("Transpose", ["xt"], ["Y"], perm=[1, 2, 0]),
                 ],
                 "name",
-                [oh.make_tensor_value_info("X", TFLOAT, [None, None, None])],
-                [oh.make_tensor_value_info("Y", TFLOAT, [None, None, None])],
+                [onnx.helper.make_tensor_value_info("X", FLOAT, [None, None, None])],
+                [onnx.helper.make_tensor_value_info("Y", FLOAT, [None, None, None])],
             ),
-            opset_imports=[oh.make_opsetid("", 18)],
+            opset_imports=[onnx.helper.make_opsetid("", 18)],
         )
 
         # back to ir
-        ir_model = oir.irbuilder.build_ir(model)
+        ir_model = ir.serde.deserialize_model(model)
 
         # starts matching
-        rule = org.make_pattern_rule(
+        rule = generic_pattern.make_pattern_rule(
             transpose_transpose_pattern,
-            transpose_transpose_apply_pattern,
+            transpose_transpose_apply_pattern(perm=[2, 0, 1]),
```

Comment:
The test does not test the same thing anymore. By defining the permutation here, it cannot be guessed from the matched nodes.

Reply:
Good call, thanks. Building patterns based on match results is support by pattern.py, could you see if that fulfills your need (as described by Rama in a separate PR)? Since the feature has been integrated to pattern.py, we removed it from generic_pattern